### PR TITLE
Mdct 451: V2 -> V3 Data Migration

### PR DESCRIPTION
**Pull request by BearHanded**
# Description

[Ticket](https://qmacbis.atlassian.net/browse/MDCT-451). The ticket also has a related spike with some extra info. Additional tables were captured in this that were not listed out in the spike.

This ticket adds a handler function to the database project that can be triggered to do a migration from v2. It uses a little migration helper I wrote that accepts objects containing instructions on how to migrate. Each object informs the runner how to handle different parts of the ETL, and are each their own file in the `database/handlers/dataMigration/tables` directory.

The command can be run locally to pull data from master for testing. On release, the "scripts" plugin in database serverless.yaml should trigger the migration if enabled.

## How to test
Look at the deployed [branch!](https://d2m9ku0tcmj6w8.cloudfront.net/)

If you want to run the migration locally, pull down the branch, setup the new env vars for migration info. 

Run `./dev local1 in one terminal to get all the lambdas stood up.

In another terminal, invoke the function locally with: 
```
 aws lambda invoke /dev/null \
  --endpoint-url http://localhost:3003 \
  --function-name database-local-dataMigration
```

The batch operations for the dynamodb client don't allow you to upsert in a batch, so each item has to be added individually. It is still pretty fast though, light database.

## Dependencies
New env config. Need to short circuit the runner when you're not using it, or have the db info configured if you plan on using it.

# Get it done
Deploy this to each env to see live data.
First pass will be run against the created branch env. Check it out [here](https://d2m9ku0tcmj6w8.cloudfront.net/).

## Code authors checklist

- [x] I have performed a self-review of my code
- [ ] I have added thorough tests. [Testing Doc](https://qmacbis.atlassian.net/wiki/spaces/CM/pages/2914025525/Test+Suite+and+Testing+Research)
- [ ] Necessary analytics were added, or no new analytics were required
- [x] I have made corresponding changes to the documentation
- [x] I have assigned the PR to myself

## Reviewers Checklist (Two different people)

- [ ] I have done the deep review and verified the items in the above checklist are g2g
- [ ] I have done the lgtm review

## Assignee

- [ ] I have closed the PR after the review and necessary changes (squashing preferred)

### CODE REVIEW

File: services/database/package.json [hunk 1]
```diff
@@ -9,6 +9,12 @@
   "author": "",
   "license": "CC0-1.0",
   "devDependencies": {
-    "serverless-dynamodb-local": "0.2.40"
+    "serverless-dynamodb-local": "0.2.40",
+    "serverless-offline": "^8.8.1",
+    "serverless-offline-ssm": "^6.2.0"
```
Comment (gmrabian) [inline services/database/package.json:14]: I think these two - offline and offline-ssm - are in the root package.json so you shouldn't need to add them here

File: services/database/package.json [hunk 2]
```diff
@@ -9,6 +9,12 @@
...
+  },
+  "dependencies": {
+    "pg": "^8.7.3",
+    "serverless-plugin-scripts": "^1.0.2"
```
Comment (gmrabian) [inline services/database/package.json:18]: same as above - plugin-scripts is in root package.json

**Comment by gmrabian** [comment]
> setup the new env vars for migration info

Can you update .env_example or share with me what I need to test it out?

**Review by gmrabian — CHANGES_REQUESTED** [review]
Changes commented above. This turned out really well all in all!

**Review by gmrabian — APPROVED** [review]

**Review by karla-vm — APPROVED** [review]

**Pull request merged**